File: DataStructuresReview/Java/src/DataStructures/Interfaces/ISingleLinkList.java
package DataStructures.Interfaces;

public interface ISingleLinkList extends IDataStructureTest {
	public void prepend(Object obj);

	public void append(Object obj);

	public void clear();

	public Object first();
	
	public Object last();	

	public void insertAfter(Object obj, Object after);

	public void insertBefore(Object obj, Object before);

	public boolean isEmpty();

	public void remove(Object obj);

	public int size();
}

File: DataStructuresReview/CSharpe/DataStructuresReview/CDataStructuresReview/CDataStructuresReview/ISingleLinkList.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CDataStructuresReview
{
    public interface ISingleLinkList<T> : 
        IDataStructureTest
    {
        void Add(T obj);

        void AddLast(T obj);

        void Clear();

        T First();

        void InsertAfter(T obj, T after);

        void InsertBefore(T obj, T before);

        bool IsEmpty();

        void Remove(T obj);

        int Size();
    }
}
